Neos.Flow/Classes/Security/Authentication/Provider/PersistedUsernamePasswordProvider.php

```diff
@@ -104,14 +104,11 @@ public function authenticate(TokenInterface $authenticationToken)
             return;
         }
 
-        if ($this->hashService->validatePassword($credentials['password'], $account->getCredentialsSource())) {
-            $account->authenticationAttempted(TokenInterface::AUTHENTICATION_SUCCESSFUL);
+        $authenticationToken->setAccount($account);
```

Comment:
This is the important subtle change I guess. I don't think it is a problem to move this before the `validatePassword()`, but I'd like to hear the others' opinion.

Reply:
Why is this change required?

Reply:
This is a dangerous change IMO because it sets the account even if the token didn't validate.
Why not just move it into the `if ($this->hashService->validatePassword(...` block?

Reply:
@bwaidelich It is require to pass the account (already found by the given accountIdentifier) to the signal for failedAuthentication. This is to remove the hardcoded "authenticationFailed" functionality of the Neos.Flow Account object and move it to a signal, so other providers is also able to handle the case of a authentication failing.

Reply:
Ah sure, I see, thanks for the explanation